construction_impl.go

```diff
@@ -15,70 +15,318 @@
+func (r RosettaConstructionFilecoin) Sign(message []byte, sk []byte) ([]byte, error) {
+	return signSecp256k1(message, sk)
+}
+
+func (r RosettaConstructionFilecoin) Verify(message []byte, publicKey []byte, signature []byte) error {
+	addr, err := address.NewSecp256k1Address(publicKey)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	return signature.MarshalBinary()
+	return verifySecp256k1(signature, addr, message)
 }
 
-func (r RosettaConstructionFilecoin) Verify(message []byte, publicKey []byte, signature []byte) error {
-	sig := crypto.Signature{}
-	err := sig.UnmarshalBinary(signature)
+func (r RosettaConstructionFilecoin) ConstructPayment(request *PaymentRequest) (string, error) {
+	to, err := address.NewFromString(request.To)
 	if err != nil {
-		return err
+		return "", err
 	}
 
-	addr, err := address.NewSecp256k1Address(publicKey)
+	from, err := address.NewFromString(request.From)
 	if err != nil {
-		return err
+		return "", err
 	}
 
-	return sigs.Verify(&sig, addr, message)
-}
+	value := types.NewInt(request.Quantity)
 
-func (r RosettaConstructionFilecoin) ConstructPayment(request *PaymentRequest) (string, error) {
-	panic("implement me")
+	gasprice, err := types.BigFromString(request.Metadata.GasPrice)
+	if err != nil {
+		return "", err
+	}
+
+	msg := &types.Message{types.MessageVersion,
+		to,
+		from,
+		request.Metadata.Nonce,
+		value,
+		gasprice,
+		request.Metadata.GasLimit,
+		builtin.MethodSend,
+		make([]byte,0),
+	}
+
+	tx, err := json.Marshal(msg)
+	if err != nil {
+		return "", err
+	}
+
+	return string(tx), nil
 }
 
 func (r RosettaConstructionFilecoin) ConstructMultisigPayment(request *MultisigPaymentRequest) (string, error) {
-	panic("implement me")
+	to, err := address.NewFromString(request.Multisig)
+	if err != nil {
+		return "", err
+	}
+
+	from, err := address.NewFromString(request.From)
+	if err != nil {
+		return "", err
+	}
+
+	value := types.NewInt(0)
+
+	gasprice, err := types.BigFromString(request.Metadata.GasPrice)
+	if err != nil {
+		return "", err
+	}
+
+	toParams, err := address.NewFromString(request.Params.To)
+	if err != nil {
+		return "", err
+	}
+
+	valueParams := types.NewInt(request.Params.Quantity)
+
+	params := &multisig.ProposeParams{
+		To: toParams,
+		Value: valueParams,
+		Method: builtin.MethodSend,
+		Params: make([]byte, 0),
+	}
+
+	buf := new(bytes.Buffer)
+	err = params.MarshalCBOR(buf)
+	serParams := buf.Bytes()
+
+	msg := &types.Message{types.MessageVersion,
+		to,
+		from,
+		request.Metadata.Nonce,
+		value,
+		gasprice,
+		request.Metadata.GasLimit,
+		builtin.MethodsMultisig.Propose,
+		serParams,
+	}
+
+	tx, err := json.Marshal(msg)
+	if err != nil {
+		return "", err
+	}
+
+	return string(tx), nil
 }
 
-func (r RosettaConstructionFilecoin) ConstructSwapAuthorizedParty(request *MultisigPaymentRequest) (string, error) {
-	panic("implement me")
+func (r RosettaConstructionFilecoin) ConstructSwapAuthorizedParty(request *SwapAuthorizedPartyRequest) (string, error) {
```

Comment:
Instead of having a `request *SwapAuthorizedPartyRequest` argument we might enumerate all the argument we need. It does make the function a bit more easy to use for third party.
Something like :
`ConstructSwapAuthorizedParty(to string, from string, quantity, ....)`
Not sure if allow ?